test/validation/api/ipsec/ipsec.c

```diff
@@ -437,6 +440,49 @@ void ipsec_sa_param_fill(odp_ipsec_sa_param_t *param,
 	param->lifetime.hard_limit.packets = 10000 * 1000;
 }
 
+static void ipsec_status_event_handle(odp_event_t ev_status,
+				      odp_ipsec_sa_t sa,
+				      enum ipsec_test_sa_expiry sa_expiry)
+{
+	odp_ipsec_status_t status = {
+		.id = 0,
+		.sa = ODP_IPSEC_SA_INVALID,
+		.result = 0,
+		.warn.all = 0,
+	};
+
+	CU_ASSERT_FATAL(ODP_EVENT_INVALID != ev_status);
+	CU_ASSERT_EQUAL(1, odp_event_is_valid(ev_status));
+	CU_ASSERT_EQUAL_FATAL(ODP_EVENT_IPSEC_STATUS, odp_event_type(ev_status));
+
+	CU_ASSERT_EQUAL(0, odp_ipsec_status(&status, ev_status));
+	CU_ASSERT_EQUAL(ODP_IPSEC_STATUS_WARN, status.id);
+	CU_ASSERT_EQUAL(sa, status.sa);
+	CU_ASSERT_EQUAL(0, status.result);
+
+	if (IPSEC_TEST_EXPIRY_IGNORED != sa_expiry) {
+		if (IPSEC_TEST_EXPIRY_SOFT_PKT == sa_expiry) {
+			CU_ASSERT_EQUAL(1, status.warn.soft_exp_packets);
+			sa_expiry_notified = true;
+		} else if (IPSEC_TEST_EXPIRY_SOFT_BYTE == sa_expiry) {
+			CU_ASSERT_EQUAL(1, status.warn.soft_exp_bytes);
+			sa_expiry_notified = true;
+		}
+	}
+
+	odp_event_free(ev_status);
+}
+
+void ipsec_status_event_get(odp_ipsec_sa_t sa,
+			    enum ipsec_test_sa_expiry sa_expiry)
+{
+	odp_event_t ev;
+
+	ev = recv_event(suite_context.queue, STATUS_EVENT_WAIT_TIME);
```

Comment:
The wait here is slowing down the inline outbound test from less than 1 second to more than 20 seconds, which is going to slow down the CI too.

I think instead of lowering the wait time, the code could perhaps try to be a bit smarter about when a nonzero wait time is needed. One think that could help is to not wait if sa_expiry is IPSEC_TEST_EXPIRY_IGNORED, but I am not sure if that is the best or only thing to do.

Reply:
In the initial proposal it was with zero wait time to mitigate this issue. It was updated as per your feedback. Are you suggesting to revert to earlier proposal?

Reply:
No I am not. This is not a binary decision but there are many different ways to do the logic.

Reply:
Updated the change to use zero wait time when sa_expiry is  IPSEC_TEST_EXPIRY_IGNORED in v12.